src/@providers/docker/services/deployment.ts

```diff
@@ -90,13 +92,33 @@ export class DockerDeploymentService extends CrudResourceService<'deployment', D
     return mergeReadableStreams(child.stdout, child.stderr);
   }
 
+  private async checkIfNetworkExists(): Promise<boolean> {
+    const { stdout, code } = await exec('docker', {
+      args: ['network', 'ls', '--format=json', '--filter', 'name=arcctl-local'],
+    });
+    if (code != 0) {
+      throw new Error('Unable to list docker networks');
+    }
+    return !!stdout;
+  }
+
+  private async createNetwork(): Promise<void> {
+    if (await this.checkIfNetworkExists()) {
+      return;
+    }
+
+    const { code } = await exec('docker', { args: ['network', 'create', DOCKER_NETWORK_NAME] });
+    if (code !== 0) {
+      throw new Error('Unable to create docker network');
+    }
+  }
+
```

Comment:
This stuff feels like you'd be better off using the namespace service within the provider.

Reply:
I was thinking about that, but this is not really a namespace or even something specific to a resource. This network is used just for our entire provider and only affects the deployment. So it felt weird to not make it the responsibility of the deployment.